plugin/trino-hive/src/main/java/io/trino/plugin/hive/fs/IOIterator.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.hive.fs;
+
+import java.io.IOException;
+
+public interface IOIterator<T>
```

Comment:
How about calling this `RemoteIterator`, then we don't need to rename everything. I also find the name `IOIterator` a bit weird.